### PR TITLE
Sort categories by order hint

### PR DESCRIPTION
**Pull request by lauraluiz**
Please review @schleichardt @peter-gerhard

### CODE REVIEW

File: app/inject/CategoryTreeProvider.java [hunk 1]
```diff
@@ -27,6 +28,7 @@ public CategoryTree get() {
         try {
             final QueryAll<Category, CategoryQuery> query = QueryAll.of(CategoryQuery.of());
             final List<Category> categories = query.run(client).toCompletableFuture().get();
+            Collections.sort(categories, (c1, c2) -> ObjectUtils.compare(c1.getOrderHint(), c2.getOrderHint()));
```
Comment (schleichardt) [inline app/inject/CategoryTreeProvider.java]: That does only work on a mutable list. Sorting a stream works on every list.

Comment (schleichardt) [inline app/inject/CategoryTreeProvider.java]: https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#sorted-java.util.Comparator-

Comment (peter-gerhard) [inline app/inject/CategoryTreeProvider.java]: for dmexco :+1:

Comment (lauraluiz) [inline app/inject/CategoryTreeProvider.java]: Nice, I didn't remember the stream option.

**Pull request merged**